packages/common/src/extensions/slickRowBasedEdit.ts

```diff
@@ -0,0 +1,596 @@
+          columns: columnDefinitions,
+          pluginName: this.pluginName,
+        });
+      }
+    }
+    return this;
+  }
+
+  getColumnDefinition(): Column {
+    const columnId = String(
+      this._addonOptions?.columnId ?? this._defaults.columnId
+    );
+
+    return {
+      id: columnId,
+      name: this._addonOptions?.actionsColumnLabel,
+      field: 'action',
+      minWidth: 70,
+      width: 75,
+      maxWidth: 75,
+      excludeFromExport: true,
+      formatter: this.actionColumnFormatter.bind(this),
+      onCellClick: this.onCellClickHandler.bind(this),
+    } as Column;
+  }
+
+  rowBasedEditCommandHandler(
+    item: any,
+    column: Column<any>,
+    editCommand: EditCommand
+  ) {
+    if (this._existingEditCommandHandler) {
+      this._existingEditCommandHandler(item, column, editCommand);
+    }
+
+    const prevSerializedValues = Array.isArray(editCommand.prevSerializedValue)
+      ? editCommand.prevSerializedValue
+      : [editCommand.prevSerializedValue];
+    const serializedValues = Array.isArray(editCommand.serializedValue)
+      ? editCommand.serializedValue
+      : [editCommand.serializedValue];
+    const editorColumns = this._gridService
+      ?.getAllColumnDefinitions()
+      .filter((col) => col.editor !== undefined);
+
+    const modifiedColumns: Column[] = [];
+    const idProperty = this._grid.getOptions().datasetIdPropertyName ?? 'id';
+    prevSerializedValues.forEach((_val, index) => {
+      const prevSerializedValue = prevSerializedValues[index];
+      const serializedValue = serializedValues[index];
+
+      if (prevSerializedValue !== serializedValue || serializedValue === '') {
+        const finalColumn = Array.isArray(editCommand.prevSerializedValue)
+          ? editorColumns?.[index]
+          : column;
+
+        if (!finalColumn) {
+          return;
+        }
+
+        this._grid.invalidate();
+        editCommand.execute();
+
+        this.renderUnsavedCellStyling(item[idProperty], finalColumn);
+        modifiedColumns.push(finalColumn);
+      }
+    });
+
+    const editedRow = this._editedRows.get(item[idProperty]);
+    const newCommands = [...(editedRow?.editCommands || [])];
+    if (modifiedColumns.length > 0) {
+      newCommands.push(editCommand);
+    }
+
+    this._editedRows.set(item[idProperty], {
+      columns: [...(editedRow?.columns || []), ...modifiedColumns],
+      editCommands: newCommands,
+      cssStyleKeys: editedRow?.cssStyleKeys || [],
+    });
+  }
+
+  protected checkOptionsRequirements(options: GridOption) {
+    if (!options?.enableCellNavigation) {
+      throw new Error(
+        `[Slickgrid-Universal] Row Based Edit Plugin requires the gridOption cell navigation (enableCellNavigation = true)`
+      );
+    }
+
+    if (!options?.editable) {
+      throw new Error(
+        `[Slickgrid-Universal] Row Based Edit Plugin requires the gridOption editable (editable = true)`
+      );
+    }
+  }
+
+  protected undoRowEdit(item: any) {
```

Comment:
actually I think this might make a great extension by itself. UndoRedoPlugin sort of which than could expose hooks and events for other plugins to consume/drive features.

that way there would be one feature-rich well tested implementation which users could also take for their own apps.

Reply:
yeah that's another good idea, future, future...